ocs_ci/deployment/aws.py

```diff
@@ -134,18 +152,245 @@ def destroy_cluster(self, log_level="DEBUG"):
             log_level (str): log level openshift-installer (default: DEBUG)
         """
         super(AWSIPI, self).destroy_cluster(log_level)
+        self.destroy_volumes()
 
-        try:
-            # Retrieve cluster name and AWS region from metadata
-            cluster_name = self.ocp_deployment.metadata.get("clusterName")
-            # Find and delete volumes
-            volume_pattern = f"{cluster_name}*"
-            logger.debug(f"Finding volumes with pattern: {volume_pattern}")
-            volumes = self.aws.get_volumes_by_name_pattern(volume_pattern)
-            logger.debug(f"Found volumes: \n {volumes}")
-            for volume in volumes:
-                self.aws.detach_and_delete_volume(
-                    self.aws.ec2_resource.Volume(volume['id'])
+
+class AWSUPI(AWSBase):
+    """
+    A class to handle AWS UPI specific deployment
+    """
+    def __init__(self):
+        self.name = self.__class__.__name__
+        super(AWSUPI, self).__init__()
+
+    class OCPDeployment(BaseOCPDeployment):
+        def __init__(self):
+            super(AWSUPI.OCPDeployment, self).__init__()
+            self.upi_repo_path = os.path.join(
+                os.path.abspath(self.cluster_path), 'openshift-misc',
```

Comment:
@dahorak what is the ENV variable that is specific to jenkins, I think its better to distinguish between jenkins and local run, either through one ENV or through cli.

Reply:
Not sure I follow here. In the jenkins we are just using some configuration like we change the bin_dir or other path which user do not usually change and using the default values. Cluster path is pointing to NFS share and so on.. But otherwise I don't see much differences between jenkins/local runs. It's just about the configuration we provide in config file and if you run on jenkins you store data like cluster dir and logs to NFS share.

Reply:
I think, this was "resolved" [above](https://github.com/red-hat-storage/ocs-ci/pull/702#discussion_r326077070).